File: hanoi.py
import matplotlib.pyplot as plt
import time


def hanoi(n, source, target, auxiliary):
    start = time.time()
    if n > 0:
        # Move n-1 disks from source to auxiliary using target as the temporary peg
        hanoi(n - 1, source, auxiliary, target)

        # Move the nth disk from source to target
        #print(f"Move disk {n} from {source} to {target}")

        # Move n-1 disks from auxiliary to target using source as the temporary peg
        hanoi(n - 1, auxiliary, target, source)
    return (time.time()-start)

# You can call the function like this:
#hanoi(10, 'A', 'C', 'B')

def main():
    tempExec = [0,0,0,0,0]
    for i in range(0,5):
        tempExec[i] = hanoi(5*i,'A','C','B')
    
    for i in range(0,5):
        print(tempExec[i])

main()

File: hanoi.c
#include <stdio.h>
#include <time.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>


double Hanoi(int n, char init, char target, char temp){

    double time_spent = 0.0;
    time_t begin = time(NULL);

    if(n > 0){
        Hanoi(n-1,init, temp, target);

        //printf("Move disk %d from %c to %c\n",n,init,target);
        Hanoi(n - 1, temp, target, init);
    }

    //sleep(1000);

    time_t end = time(NULL);

    time_spent += end - begin;
    return time_spent;
}


int main() {
    double tempExec[5] = {0,0,0,0,0};
    for (int i = 0; i<5;i++){
        tempExec[i] = Hanoi(5*i,'A','C','B');
    }
    for (int i = 0; i<5;i++){
        printf("/%f/",tempExec[i]);
    }

    printf("\n%f",Hanoi(10,'A','C','B'));
    return 0;
}
